test/shared_utilities/file_reader.jl

```diff
@@ -113,51 +245,68 @@ if !Sys.iswindows()
         surface_space_t = Spaces.SpectralElementSpace2D(topology, quad)
         data_fields = (zeros(surface_space_t), ones(surface_space_t))
 
+        # Add 2 variables to `FileInfo`, but only read/write `var1`
+        varnames = ["var1", "var2"]
+
         # create structs manually since we aren't reading from a data file
         file_info = FileReader.FileInfo(
             "",             # infile_path
             "",             # regrid_dirpath
-            "",             # varname
+            varnames,       # varnames
             "",             # outfile_root
             dummy_dates,    # all_dates
             date_idx0,      # date_idx0
         )
-        file_state = FileReader.FileState(
+        file_state1 = FileReader.FileState(
+            data_fields,        # data_fields
+            copy(date_idx0),    # date_idx
```

Comment:
I guess date_idx and segment should be the same for each variable's FileState, so there is some duplicate into here. but it may be easier to store them for each? because then FileState is self contained.

Reply:
Hmm yeah good point. It is really nice to have each `file_state` self-contained - we take advantage of this in `get_data_at_date` and it makes it easy to loop over `varname, file_state` pairs in `read_data_fields!`